src/datahike/db.cljc

```diff
@@ -367,34 +368,62 @@
   (-index-range [db attr start end]
                 (filter (.-pred db) (-index-range (.-unfiltered-db db) attr start end))))
 
+(defn multi-comp
+  "Based on https://clojuredocs.org/clojure.core/sort#example-58e81eb1e4b01f4add58fe88"
+  ([fns a b]
+   (multi-comp fns < a b))
+  ([[f & others :as fns] order a b]
+   (if (seq fns)
+     (let [result (kc/-compare (f a) (f b))
+           f-result (if (= order >) (* -1 result) result)]
+       (if (= 0 f-result)
+         (recur others order a b)
+         f-result))
+     0)))
+
+(defn concat-sort [list-a list-b index-type]
```

Comment:
How efficient is this fn?

Reply:
The algorithmic complexity of this algorithm is the length of the longest list in {list-a, list-b}, which is better than resorting everything (which is O(n*log n) and optimal here. If all of the output is expected to be consumed it can be more efficient to remove the laziness, but I guess sometimes only part of it will be consumed, right?